rust/template/tcp_channel/receiver.rs

```diff
@@ -122,14 +198,22 @@ where
 
 impl<T> Drop for TcpReceiver<T> {
     fn drop(&mut self) {
-        // We can't really handle any errors here.
-        // The order of operations is important. First close the
-        // listener to be sure that a no new stream will be accepted,
-        // then close the stream itself.
-        unsafe {
-            let _ = close(self.listener);
-            let _ = self.stream.lock().unwrap().map(|fd| close(fd));
+        {
+            let mut guard = self.data.lock().unwrap();
+            guard.closed = true;
+            // TODO: We probably want to just log failures.
+            guard.fd.close().unwrap();
         }
+
+        // TODO: We probably want to log any errors reported by the
+        //       thread being joined.
+        self.thread
+            .take()
+            .map(JoinHandle::join)
+            .unwrap()
```

Comment:
Won't this crash if the receiver thread returns an error?

Reply:
Yeah. For now I concluded that this is better than not having any signalling that something went wrong (which was the cause of a long debug session already). I'll change this to behave more friendly once I added logging.

Reply:
As long as there is a big FIXME in the code :)